Scalar.Common/Maintenance/ConfigStep.cs

```diff
@@ -209,43 +209,59 @@ public bool TrySetConfig(out string error)
                 return false;
             }
 
+            // `feature.scalar` is a meta-variable which might control many
+            //    individual config settings.  It currently controls use of
+            //    an FSMonitor helper.
+            //
+            //    "false"        : disable FSMonitor completely
+            //    "true"         : use Watchman
+            //    "experimental" : use builtin FSMonitor daemon
+
             GitProcess.ConfigResult config = null;
             GitProcess.Result getResult = this.RunGitCommand(
                 process => {
-                    config = process.GetFromLocalConfig("feature.scalar");
+                    config = process.GetFromConfig("feature.scalar");
                     return null;
                 },
-                nameof(GitProcess.GetFromLocalConfig)
+                nameof(GitProcess.GetFromConfig)
             );
             GitFeatureFlags flags = GitVersion.GetAvailableGitFeatures(this.Context.Tracer);
-            config.TryParseAsString(out string scalar, out error, defaultValue: "true");
+            config.TryParseAsString(out string scalar, out error, defaultValue: "false");
 
             if (StringComparer.OrdinalIgnoreCase.Equals(scalar, "false"))
             {
-                GitProcess.Result deleteResult = this.RunGitCommand(
-                    process => process.DeleteFromLocalConfig("core.fsmonitor"),
-                    nameof(GitProcess.DeleteFromLocalConfig)
-                );
+                DisableWatchmanIntegration();
+                DisableBuiltinFSMonitorIntegration();
 
-                return deleteResult.ExitCodeIsSuccess;
+                return true;
             }
-            else if (StringComparer.OrdinalIgnoreCase.Equals(scalar, "experimental")
-                     // Make sure Git supports builtin FS Monitor
-                     && flags.HasFlag(GitFeatureFlags.BuiltinFSMonitor))
+            else if (StringComparer.OrdinalIgnoreCase.Equals(scalar, "experimental"))
             {
-                // ":internal:" is a custom value to specify the builtin
-                // FS Monitor feature.
-                GitProcess.Result setResult = this.RunGitCommand(
-                    process => process.SetInLocalConfig("core.fsmonitor", ":internal:"),
-                    nameof(GitProcess.SetInLocalConfig)
-                );
-
-                return setResult.ExitCodeIsSuccess;
+                DisableWatchmanIntegration();
+
+                if (flags.HasFlag(GitFeatureFlags.BuiltinFSMonitor))
+                {
+                    return ConfigureBuiltinFSMonitorIntegration();
+                }
+                else
+                {
+                    error = $"Builtin FSMonitor not supported by this version of Git";
+                    this.Context.Tracer.RelatedError(error);
+                    return false;
+                }
```

Comment:
This [causes the Ubuntu builds to break](Error: Failed to set initial config: Builtin FSMonitor not supported by this version of Git).

This should just be a warning, then continue to the Watchman integration (if available).

Reply:
@derrickstolee  Do we want to fall back to watchman here or do we want to amend the test matrix in the .yml to not spawn those cases at all ??

Reply:
I think for now, i'll go with your suggestion and fallbakc to Watchman if available.  I think longer term we want the `feature.scalar=experimental` to evolve and test other things.

Reply:
Yeah, and the builds won't have Watchman installed, anyway, so they'll be wasted, but :shrug:.